src/plugins/input_control_vis/public/control/range_control_factory.ts

```diff
@@ -134,15 +134,21 @@ export async function rangeControlFactory(
   deps: InputControlVisDependencies
 ): Promise<RangeControl> {
   const [, { data: dataPluginStart }] = await deps.core.getStartServices();
-  const indexPattern = await dataPluginStart.indexPatterns.get(controlParams.indexPattern);
+
+  let indexPattern;
+  try {
+    indexPattern = await dataPluginStart.indexPatterns.get(controlParams.indexPattern);
+  } catch (e) {
+    // invalid index pattern id
+  }
 
   return new RangeControl(
     controlParams,
     new RangeFilterManager(
       controlParams.id,
       controlParams.fieldName,
-      indexPattern,
-      deps.data.query.filterManager
+      deps.data.query.filterManager,
```

Comment:
@DziyanaDzeraviankina  I thought about your PR a little, your solution works as expected but what do you think if we do something like: 

 ```
    new RangeFilterManager(
      controlParams.id,
      controlParams.fieldName,
      controlParams. indexPattern,
      dataPluginStart.indexPatterns,
      deps.data.query.filterManager
    ),
  ```
it allow us to get a  specific `indexPattern` on demand instead of getting it in factory method.